rust_toolchain/lib/dependabot/rust_toolchain/requirement.rb

```diff
@@ -0,0 +1,232 @@
+# typed: strong
+# frozen_string_literal: true
+
+require "dependabot/requirement"
+require "dependabot/utils"
+
+require "dependabot/rust_toolchain/version"
+require "dependabot/rust_toolchain/channel_parser"
+require "dependabot/rust_toolchain/channel_type"
+
+module Dependabot
+  module RustToolchain
+    class Requirement < Dependabot::Requirement
+      # For consistency with other languages, we define a requirements array.
+      # rust-toolchain always contains a single element.
+      sig { override.params(requirement_string: T.nilable(String)).returns(T::Array[Dependabot::Requirement]) }
+      def self.requirements_array(requirement_string)
+        [new(requirement_string)]
+      end
+
+      sig do
+        params(obj: T.any(String, Gem::Version)).returns(T::Array[T.any(String, Dependabot::RustToolchain::Version)])
+      end
+      def self.parse(obj)
+        return ["=", RustToolchain::Version.new(obj.to_s)] if obj.is_a?(Gem::Version)
+
+        requirement_string = obj.to_s.strip
+
+        # Parse requirement strings with operators (e.g., ">= 1.72.0")
+        match = requirement_string.match(/^(>=|>|<=|<|=)\s?(.+)$/)
```

Comment:
In other ecosystems I've seen 'bump' syntax operator `~>` that essentially allows only the last listed number to change.  E>g., `~>1.72.0` would allow an update to `1.72.5` but not to `1.73.0`.  Similarly, `~>1.72` would allow an update to `1.75` but not to `2.0`.

Reply:
In what cases though? `rust-toolchain.toml` expects an exact version, not a range. I only implemented this to support `ignore-conditions` in Dependabot configuration.